tests/integration/test_agentd/test_agentd_multi_server/test_agentd_multi_server.py

```diff
@@ -0,0 +1,204 @@
+        - test_configuration:
+            type: data
+            brief: Configuration used in the test.
+        - test_metadata:
+            type: data
+            brief: Configuration cases.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Set internal configuration for testing.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Reset the 'ossec.log' file and start a new monitor.
+        - remove_keys_file:
+            type: fixture
+            brief: Deletes keys file if test configuration request it    
+
+    assertions:
+        - Agent without keys. Verify that all servers will refuse the connection to the 'wazuh-remoted' daemon
+          but will accept enrollment. The agent should try to connect and enroll each of them.
+        - Agent without keys. Verify that the first server only has enrollment available, and the third server
+          only has the 'wazuh-remoted' daemon available. The agent should enroll in the first server and
+          connect to the third one.
+        - Agent without keys. Verify that the agent should enroll and connect to the first server, and then
+          the first server will disconnect. The agent should connect to the second server with the same key.
+        - Agent without keys. Verify that the agent should enroll and connect to the first server, and then
+          the first server will disconnect. The agent should try to enroll in the first server again,
+          and then after failure, move to the second server and connect.
+        - Agent with keys. Verify that the agent should enroll and connect to the last server.
+        - Agent with keys. Verify that the first server is available, but it disconnects, and the second and
+          third servers are not responding. The agent on disconnection should try the second and third servers
+          and go back finally to the first server.
+
+    input_description: An external YAML file (wazuh_conf.yaml) includes configuration settings for the agent.
+                       Different test cases are found in the test module and include parameters for
+                       the environment setup, the requests to be made, and the expected result.
+
+    expected_output:
+        - r'Requesting a key from server'
+        - r'Valid key received'
+        - r'Trying to connect to server'
+        - r'Connected to enrollment service'
+        - r'Received message'
+        - r'Unable to connect to enrollment service at'
+
+    tags:
+        - simulator
+        - ssl
+        - keys
+    '''
+
+    # Servers paremeters
+    remoted_server_addresses = ["127.0.0.0","127.0.0.1","127.0.0.2"]
+    remoted_server_ports = [1514,1516,1517]
+    remoted_servers = [None,None,None]
+    
+    # Stop target Agent
+    control_service('stop')
```

Comment:
Why isn't this a fixture?

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/18353/commits/ecf23d03eafffc97b67a2d364324e916eb50d143